constraintlayout/constraintlayout/src/main/java/androidx/constraintlayout/helper/widget/Grid.java

```diff
@@ -1010,4 +1010,14 @@ public void setVerticalGaps(float verticalGaps) {
         generateGrid(true);
         invalidate();
     }
+
+    @SuppressLint("WrongCall")
+    @Override
+    protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
+        ConstraintLayout.LayoutParams params = params(this);
+        setMeasuredDimension(mContainer.getMeasuredWidth()
+                        - params.leftMargin - params.rightMargin
+                , mContainer.getMeasuredHeight()
+                        - params.topMargin - params.bottomMargin);
+    }
```

Comment:
Maybe we should not have this and just 
set  mUseViewMeasure  in init
 @Override
    protected void init(AttributeSet attrs) {
        super.init(attrs);
       
       mUseViewMeasure = false;
       ...